File: ISA/src/main/java/com/example/ISA/model/Patient.java
package com.example.ISA.model;
import java.util.*;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;


@Entity
public class Patient extends User {
	
	private static final long serialVersionUID = 1L;

	@Column
    private BloodType bloodType;
    
    @OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private Set<Complaint> complaints;
    
    @Column
    private int points;
    
    @Column
    private CategoryType category;
    

  

	public Patient() {
		super();
	}
	
	public Patient(int id, String email, String password, String firstName, String lastName, String address, String city, String country,
			 String phoneNumber, String jmbg, GenderType gender, String job, String info, boolean enabled, String role, List<Authority> authorities) {
		super();
		setId(id);
		setEmail(email);
		setPassword(password);
		setFirstName(firstName);
		setLastName(lastName);
		setAddress(address);
		setCity(city);
		setCountry(country);
		setPhoneNumber(phoneNumber);
		setJmbg(jmbg);
		setGender(gender);
		setJob(job);
		setInfo(info);
		setEnabled(enabled);
		setRole(role);
		setAuthorities(authorities);
	}

	public Patient(BloodType bloodType, Set<Complaint> complaints, int points, CategoryType category) {
		super();
		this.bloodType = bloodType;
		this.complaints = complaints;
		this.points = points;
		this.category = category;
	}

	public BloodType getBloodType() {
		return bloodType;
	}

	public void setBloodType(BloodType bloodType) {
		this.bloodType = bloodType;
	}

	public Set<Complaint> getComplaints() {
		return complaints;
	}

	public void setComplaints(Set<Complaint> complaints) {
		this.complaints = complaints;
	}

	public int getPoints() {
		return points;
	}

	public void setPoints(int points) {
		this.points = points;
	}

	public CategoryType getCategory() {
		return category;
	}

	public void setCategory(CategoryType category) {
		this.category = category;
	}



    
    

}

File: ISA-front/src/app/model/patient.ts
import { Address } from "./address";



interface PatientInterface{

    id:number;
    email:string;
    password1:string;
    password2:string;
    firstName:string;
    lastName:string;
    address:string;
    city:string;
    country:string;
    phoneNumber:string;
    jmbg:string;
    gender:string;
    job:string;
    info:string;
    isActivated:boolean;

    //bloodType:string;
    //points:number;
    //penalties:number;
    //complaints:Complaint[];
    //category:string;


    
}
export class Patient implements PatientInterface{
    id:number;
    email:string;
    password1:string;
    password2:string;
    firstName:string;
    lastName:string;
    address:string;
    city:string;
    country:string;
    phoneNumber:string;
    jmbg:string;
    gender:string;
    job:string;
    info:string;
    isActivated:boolean;

    //bloodType:string;
    //points:number;
    //penalties:number;
    //complaints:Complaint[];
    //category:string;


    constructor(obj:PatientInterface){
        this.id=obj.id;
        this.email=obj.email;
        this.password1=obj.password1;
        this.password2=obj.password2;
        this.firstName=obj.firstName;
        this.lastName=obj.lastName;
        this.address=obj.address;
        this.city=obj.city;
        this.country=obj.country;
        this.phoneNumber=obj.phoneNumber;
        this.jmbg=obj.jmbg;
        this.gender=obj.gender;
        this.job=obj.job;
        this.info=obj.info;
        this.isActivated=obj.isActivated;
        //this.bloodType=obj.bloodType;
        //this.points=obj.points;
        //this.penalties=obj.penalties;
        //this.category=obj.category;


    } 
}